client/client-app.cpp

```diff
@@ -0,0 +1,62 @@
+#include "client.h"
+#include "posix_wrapping.h"
+
+#include <stdexcept>    // dedault exceptions;
+#include <unistd.h>     // STDOUT_FINO;
+#include <string>
+#include <iostream>     // cout, getline, cin, endl;
+
+void singleMessageMode(cs::Client& client, std::string& msg)
+{
+    client.sendMessage(msg);
+
+    std::string buffer;
+    client.readMessage(buffer);
+    buffer.push_back('\n');
+
+    std::cout << buffer << std::endl;
+}
+
+void interactiveMode(cs::Client& client)
+{
+    std::string buffer;
+
+    while (true)
+    {
+        buffer.clear();
+        printf("Waiting for your message: ");
+        std::getline(std::cin, buffer);     // read full line;
+        if (buffer == "\n" || buffer == "") 
+            break;
+
+        singleMessageMode(client, buffer);
+    }
+}
+
+int main(int argc, char **argv)
```

Comment:
write this on the begin of the file and put functions below

Reply:
Done